### DISCUSSION TITLE
move_semantics1.rs

### DISCUSSION THREAD

**Discussion by Kaiskii** [rust-lang/rustlings#2162] [Q&A] [answered]
Hello, I am on move_semantics1.rs and this was the solution I came to:

```rs
// TODO: Fix the compiler error in this function.
fn fill_vec(mut vec: Vec<i32>) -> Vec<i32> {
    vec.push(88);

    vec
}
```

Since the TODO was to just fix the compiler error, I was assuming that this is fine, but when I checked out the solution, it uses the `shadowing` concept instead. Which makes it so that the `vec` param passed in won't be considered mutable and instead it has the `let vec` be mutable.

```rs
fn fill_vec(vec: Vec<i32>) -> Vec<i32> {
    let mut vec = vec;
    //  ^^^ added

    vec.push(88);

    vec
}
```

While I understand that the two solutions are different, I am having a hard time understanding the situation where you would want to shadow the `vec` instead of having it be mutable in params since it's more straightforward to do so.

**Comment by mo8it** [comment, marked as answer]
This exercise demonstrates that you can declare something as mutable when you own it, even if it was immutable.

`move_semantics3` recommends exactly what you did by adding `mut` to the parameter. This is the idiomatic way.

**Reply by Kaiskii** [reply]
Neat, thanks for explaining that this concept was later explained in a separate exercise!
